src/commands/detectorDiagnostics.ts

```diff
@@ -0,0 +1,130 @@
+import * as vscode from 'vscode';
+import * as k8s from 'vscode-kubernetes-tools-api';
+import { IActionContext } from "vscode-azureextensionui";
+import { AppLensARMresponse, AppLensAPIResult } from './models/applensarmresponse';
+import * as path from 'path';
+import * as fs from 'fs';
+import * as htmlhandlers from "handlebars";
+import { Errorable } from './utils/errorable';
+import ResourceManagementClient from 'azure-arm-resource/lib/resource/resourceManagementClient';
+
+const meta = require('../../package.json');
+
+export default async function detectorDiagnostics(
+    context: IActionContext,
+    target: any
+  ): Promise<void> {
+    const cloudExplorer = await k8s.extension.cloudExplorer.v1;
+
+    if (cloudExplorer.available) {
+      const cloudTarget = cloudExplorer.api.resolveCommandTarget(target);
+
+      if (cloudTarget && cloudTarget.cloudName === "Azure" &&
+            cloudTarget.nodeType === "resource" && cloudTarget.cloudResource.nodeType === "cluster") {
+              const extensionPath = getExtensionPath();
+              if (extensionPath) {
+                loadClusterInsights(cloudTarget, extensionPath);
+              }
+        } else {
+          vscode.window.showInformationMessage('This command only applies to AKS clusters.');
+        }
+    }
+}
+
+function getExtensionPath(): string | undefined {
```

Comment:
Honestly I still feel like you are better getting the `ExtensionContext` in here - but given that this works maybe it's not worth the effort!

Reply:
:bulb: Discussed and we will get on with the `getExtension` vscode function.